fitbenchmarking/core/tests/test_results_output.py

```diff
@@ -77,29 +78,30 @@ def tearDown(self):
         """
         shutil.rmtree(self.results_dir)
 
-    def test_save_results_correct_output_path(self):
+    @patch("fitbenchmarking.core.results_output.create_plots")
+    @patch("fitbenchmarking.core.results_output.fitting_report.create")
+    @patch("fitbenchmarking.core.results_output.problem_summary_page.create")
+    @patch("fitbenchmarking.core.results_output.tables.create_results_tables")
+    @patch("fitbenchmarking.core.results_output.create_problem_level_index")
+    def test_save_results(self, m1, m2, m3, m4, m5):
         """
-        Tests to check the group_dir is correct
+        Tests to check the group_dir is correct and a dict of
+        pandas dataframe is returned
         """
-        failed_problems = []
-        unselected_minimizers = {}
-        group_name = "group_name"
-        group_dir, _ = save_results(self.options, self.results,
-                                    group_name, failed_problems,
-                                    unselected_minimizers)
-        self.assertEqual(group_dir, os.path.join(self.results_dir, group_name))
+        m1.return_value = None
+        m2.return_value = (None, None)
+        m3.return_value = None
+        m4.return_value = None
+        m5.return_value = None
 
-    def test_save_results_correct_output_dict_format(self):
-        """
-        Test to check save_results returns a dict of pandas dataframes.
-        """
         failed_problems = []
         unselected_minimizers = {}
         group_name = "group_name"
-        _, pp_dfs = save_results(self.options, self.results,
-                                 group_name, failed_problems,
-                                 unselected_minimizers)
+        group_dir, pp_dfs = save_results(self.options, self.results,
+                                         group_name, failed_problems,
+                                         unselected_minimizers)
 
+        self.assertEqual(group_dir, os.path.join(self.results_dir, group_name))
         assert isinstance(pp_dfs, dict)
         for df in list(pp_dfs.values()):
             assert isinstance(df, pd.DataFrame)
```

Comment:
These two tests were failing when I made the changes to the tables and removed the links (I later figured out why it was failing and that is why an additional guard clause was added to a method in the local_min_table). I added the mocks because for what the tests are testing it does not really need to call the other functions. After adding the mocks to both the tests, I saw that they were basically doing the same thing and I could merge them into a single function to test both the return values. I thought it was cleaner to do it this way.